### DISCUSSION TITLE
Generating an HTML "shell" that can feed off of a previous goaccess outputted JSON file.

### DISCUSSION THREAD

**Discussion by sedevelops** [allinurl/goaccess#2593] [Q&A]
Hi!

Great product - thank you for sharing it with us!  My question:

I'm up and running with GoAccess and loving it. It's all that I need for stats. Perfecto.

I have set up my apache access logs to rotate at midnight each night via cron job. Then, within that same cron job, I task goaccess with parsing the previous day's log file and saving a JSON file to disk.

I don't have a need to monitor my stats all the time, but will want to access them on occasion if there's a secuirity event or other interesting traffic pattern I want to understand. So I'd like to just keep an archive of these daily parsed JSON files.

My question is: Does anyone know if I can retroactively task goaccess with producing an html "shell" to feed off of these old parsed JSON files at a later date?

Basically, is there a way that goaccess can generate this html shell for me and I can simply edit within it to point to the location of a previously 'dumped' JSON file?

**Comment by allinurl** [comment]
Certainly! Great question. Currently, loading GoAccess' JSON output to generate an HTML report isn't a feature. However, I find this to be an excellent idea. Implementing it might not be too complex. It would likely involve adding a new command line option and modifying the process to load the JSON file and pass it to the [`print_json_data`](https://github.com/allinurl/goaccess/blob/master/src/output.c#L508)() function in output.c. 

Are you familiar with C? If you're interested, I'd gladly review and merge your pull request!

**Reply by sedevelops** [reply]
Wow!  Sorry I didn't see this sooner.  And I'm sorry too that I'm not familiar at ALL with C.  I'm all in on PHP, JS and other web dev areas. But clueless beyond that for the time being.

Also, thank you so much for maintaining such a great tool.

**Reply by allinurl** [reply]
I can explore adding this option. For now, an alternative approach is to generate the HTML file with each cron job and archive those. If you require the JSON version later for a specific report, the HTML report offers an export option to JSON.
